x-pack/plugins/ml/public/jobs/components/custom_url_editor/editor.js

```diff
@@ -39,17 +41,32 @@ import {
 function getLinkToOptions() {
   return [{
     id: URL_TYPE.KIBANA_DASHBOARD,
-    label: 'Kibana dashboard',
+    label: i18n.translate('xpack.ml.customUrlEditor.kibanaDashboardLabel', {
+      defaultMessage: 'Kibana dashboard'
+    }),
   }, {
     id: URL_TYPE.KIBANA_DISCOVER,
-    label: 'Discover',
+    label: i18n.translate('xpack.ml.customUrlEditor.discoverLabel', {
+      defaultMessage: 'Discover'
+    }),
   }, {
     id: URL_TYPE.OTHER,
-    label: 'Other',
+    label: i18n.translate('xpack.ml.customUrlEditor.otherLabel', {
+      defaultMessage: 'Other'
+    }),
   }];
 }
 
-export class CustomUrlEditor extends Component {
+export const CustomUrlEditor = injectI18n(class extends Component {
+  static DisplayName = 'CustomUrlEditor';
```

Comment:
do not use displayName(also displayName should start with lowercase). Use class CustomUrlEditor extends Component instead